b2sdk/transfer/emerge/executor.py

```diff
@@ -329,6 +342,11 @@ def _find_unfinished_file_by_plan_id(
                 # pass UNKNOWN file_retention here - but raw_api/server won't allow it
                 # and we don't check it here
                 continue
+
+            if cache_control is None or file_.cache_control != encryption:
```

Comment:
```suggestion
            if cache_control is None or file_.cache_control != cache_control:
```
A test should have found that.
This slipping through to a pull request shows that we are missing an important test.